### PR TITLE
RLD-20 Inputs + Form Fields

### PR DESCRIPTION
**Pull request by ainouzgali**

### CODE REVIEW

File: apps/web/src/design-system/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+} as ComponentMeta<typeof TextInput>;
+
+const Template: ComponentStory<typeof TextInput> = ({ ...args }) => <TextInput {...args} />;
+
+export const PrimaryUse = Template.bind({});
+PrimaryUse.args = {
+  label: 'Notification Name',
+  placeholder: 'Notification name goes here...',
+};
+
```
Comment (scopsy) [inline apps/web/src/design-system/TextInput/TextInput.stories.tsx:24]: Also missing the description prop here:
CleanShot 2022-02-08 at 13 00 47@2x

**Review by scopsy — COMMENTED** [review]
@ainouzgali Pixel perfect! 🤩 A couple of comments and missing states

File: apps/web/src/design-system/TextInput/TextInput.styles.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { createStyles, MantineTheme } from '@mantine/core';
+
+export default createStyles((theme: MantineTheme) => {
```
[the remark this answers is not in the document]
Comment (ainouzgali) [reply apps/web/src/design-system/TextInput/TextInput.styles.ts:3]: I created a different issue for that on linear under Icons. There is definitely a manine option for the input sides and icons.

File: apps/web/src/design-system/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  error?: string;
+  placeholder?: string;
+  value?: string;
+}
+
+/**
+ * Text Input component
+ *
+ */
+export function TextInput({ ...props }: ITextInputProps) {
```
[the remark this answers is not in the document]
Comment (ainouzgali) [reply apps/web/src/design-system/TextInput/TextInput.tsx:16]: This TextInput could receive a type of number\text\email\url etc. Or there is a mantine NumberInput with extra number-related functionality.  
But for now I think you are right , should be renamed  Input.

File: apps/web/src/design-system/TextInput/TextInput.stories.tsx (hunk 1 above)
Comment (ainouzgali) [reply apps/web/src/design-system/TextInput/TextInput.stories.tsx:24]: description added

**Review by scopsy — APPROVED** [review]
Mushhhhh 😎 Ship it!!! :)

**Pull request merged**